ir/visitor.h

```diff
@@ -165,7 +165,10 @@ class Visitor {
     virtual bool join_flows(const IR::Node *) { return false; }
     void visit_children(const IR::Node *, std::function<void()> fn) { fn(); }
     class ChangeTracker;  // used by Modifier and Transform -- private to them
-#ifndef NDEBUG
+#ifdef NDEBUG
+#define override_unless_NDEBUG
+#else
+#define override_unless_NDEBUG  override
```

Comment:
The macro should probably have `NDEBUG` in its name somewhere, since its specific to compiling with NDEBUG to disable assertion checks.  Or just take out the whole thing (make the check unconditional) and don't worry about the efficiency.